safe/impact_function/impact_function.py

```diff
@@ -436,6 +436,12 @@ def outputs(self):
 
         # Remove layers which are not set.
         layers = [layer for layer in layers.values() if layer]
+
+        if len(layers) != self._output_number():
+            # If this happen, it's an error from developpers.
+            raise Exception(
```

Comment:
Perhaps give more information, like how many expected layers and what it gets, to see where is the problem.
Even better, if we can also show which layer is missing (but it seems your current code doesn't support this).